In [ ]:
import pandas as pd

# Cargar el archivo CSV
df_netflix_edaproducciones = pd.read_csv(r"C:\Users\fatim\Downloads\laboratorio-Pandas_Netflixproducciones\data.netflix.csv") 

# Configurar Pandas para mostrar todas las filas y columnas de nuestro conjunto de datos
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

# Mostrar los DataFrame completos a través de display(df_netflix_edaproducciones) para darle un formato más visual que el que devuelve print
print("#### Dataframe de todas las producciones de Netflix ####")
display(df_netflix_edaproducciones_info = {
    "Head": df_netflix_edaproducciones.head(),  # First few rows of the dataset
    "Shape": df_netflix_edaproducciones.shape,  # Dimensions of the dataset
    "Columns": df_netflix_edaproducciones.columns.tolist(),  # List of column names
    "Summary": df_netflix_edaproducciones.describe(include='all')  # Resumen estadístico de nuestras variables numéricas y categóricas
})

display(df_netflix_edaproducciones_info)




#### Dataframe de todas las producciones de Netflix ####


{'Head':    Unnamed: 0 show_id     type                  title         director  \
 0           0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
 1           1      s2  TV Show          Blood & Water          Unknown   
 2           2      s3  TV Show              Ganglands  Julien Leclercq   
 3           3      s4  TV Show  Jailbirds New Orleans          Unknown   
 4           4      s5  TV Show           Kota Factory          Unknown   
 
                                                 cast        country  \
 0                                            Unknown  United States   
 1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
 2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...        Unknown   
 3                                            Unknown        Unknown   
 4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
 
            date_added  release_year rating   duration  \
 0  September 25, 2021          2020  PG-13     

In [21]:

# Resumen general
print("Resumen de columnas:")
display(df_netflix_edaproducciones.info())

# Valores únicos en las principales columnas
print("\nValores únicos en 'type':")
display(df_netflix_edaproducciones['type'].value_counts())

print("\nDistribución por país:")
display(df_netflix_edaproducciones['country'].value_counts().head(10))

# Valores nulos
print("\nValores faltantes por columna:")
display(df_netflix_edaproducciones.isnull().sum())

# Resumen estadístico 
print("\nResumen estadístico de IMDb Score:")
display(df_netflix_edaproducciones['IMDB Score'].describe())

# Limpieza básica de datos
df_netflix_edaproducciones['country'] = df_netflix_edaproducciones['country'].replace('Unknown', pd.NA)

# Filtrar películas con alta calificación
high_rated = df_netflix_edaproducciones[df_netflix_edaproducciones['IMDB Score'] > 8]
print("\nPelículas con calificación IMDb mayor a 8:")
display(high_rated[['title', 'IMDB Score', 'country']])



Resumen de columnas:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 20 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Unnamed: 0    513 non-null    int64  
 1   show_id       513 non-null    object 
 2   type          513 non-null    object 
 3   title         513 non-null    object 
 4   director      513 non-null    object 
 5   cast          513 non-null    object 
 6   country       341 non-null    object 
 7   date_added    513 non-null    object 
 8   release_year  513 non-null    int64  
 9   rating        513 non-null    object 
 10  duration      513 non-null    object 
 11  listed_in     513 non-null    object 
 12  description   513 non-null    object 
 13  Title         513 non-null    object 
 14  Genre         513 non-null    object 
 15  Premiere      513 non-null    object 
 16  Runtime       513 non-null    int64  
 17  IMDB Score    513 non-null    float64
 18  Language 

None


Valores únicos en 'type':


type
Movie      337
TV Show    176
Name: count, dtype: int64


Distribución por país:


country
United States                    125
India                             36
Japan                             22
United Kingdom                    13
Nigeria                           10
United States, Canada             10
United Kingdom, United States      6
France                             5
Australia                          5
United States, United Kingdom      5
Name: count, dtype: int64


Valores faltantes por columna:


Unnamed: 0        0
show_id           0
type              0
title             0
director          0
cast              0
country         172
date_added        0
release_year      0
rating            0
duration          0
listed_in         0
description       0
Title             0
Genre             0
Premiere          0
Runtime           0
IMDB Score        0
Language          0
is_original       0
dtype: int64


Resumen estadístico de IMDb Score:


count    513.000000
mean       6.210916
std        0.968850
min        2.500000
25%        5.600000
50%        6.300000
75%        6.900000
max        9.000000
Name: IMDB Score, dtype: float64


Películas con calificación IMDb mayor a 8:


title  IMDB Score  \
502            Dennis the Menace         8.1   
503                  Dreamy Eyes         8.1   
504             Dynasty Warriors         8.1   
505                Kung Fu Panda         8.2   
506              Kung Fu Panda 2         8.2   
507           Life as We Know It         8.2   
508               Mary Magdalene         8.3   
509              Masameer County         8.3   
510  Mobile Suit Gundam Hathaway         8.4   
511                Mortal Kombat         8.5   
512          No Strings Attached         9.0   

                                      country  
502                             United States  
503                                   Vietnam  
504                                      <NA>  
505                      United States, China  
506                      United States, China  
507                             United States  
508  United Kingdom, Australia, United States  
509                                      <NA>  
510                                      <NA>  
511                             United States  
512             United States, United Kingdom

In [24]:
print("Todas las estadísticas descriptivas de las producciones de Netflix")
display(df_netflix_edaproducciones.describe(include='all').transpose()) #todas las columnas

display(df_netflix_edaproducciones.describe().T) #solo categóricas

Todas las estadísticas descriptivas de las producciones de Netflix


count unique                                                top  \
Unnamed: 0    513.0    NaN                                                NaN   
show_id         513    513                                               s584   
type            513      2                                              Movie   
title           513    513                                No Strings Attached   
director        513    310                                            Unknown   
cast            513    443                                            Unknown   
country         341     86                                      United States   
date_added      513     74                                  September 1, 2021   
release_year  513.0    NaN                                                NaN   
rating          513     10                                              TV-MA   
duration        513     78                                            No data   
listed_in       513    165                           Children & Family Movies   
description     513    510  Paranormal activity at a lush, abandoned prope...   
Title           513    513           David Attenborough: A Life on Our Planet   
Genre           513    106                                        Documentary   
Premiere        513    350                                   October 18, 2019   
Runtime       513.0    NaN                                                NaN   
IMDB Score    513.0    NaN                                                NaN   
Language        513     37                                            English   
is_original     513      1                                               True   

             freq         mean         std     min     25%     50%     75%  \
Unnamed: 0    NaN   280.984405  166.445537     0.0   137.0   277.0   422.0   
show_id         1          NaN         NaN     NaN     NaN     NaN     NaN   
type          337          NaN         NaN     NaN     NaN     NaN     NaN   
title           1          NaN         NaN     NaN     NaN     NaN     NaN   
director      158          NaN         NaN     NaN     NaN     NaN     NaN   
cast           58          NaN         NaN     NaN     NaN     NaN     NaN   
country       125          NaN         NaN     NaN     NaN     NaN     NaN   
date_added     54          NaN         NaN     NaN     NaN     NaN     NaN   
release_year  NaN  2014.181287    9.995716  1959.0  2010.0  2019.0  2021.0   
rating        159          NaN         NaN     NaN     NaN     NaN     NaN   
duration      300          NaN         NaN     NaN     NaN     NaN     NaN   
listed_in      22          NaN         NaN     NaN     NaN     NaN     NaN   
description     4          NaN         NaN     NaN     NaN     NaN     NaN   
Title           1          NaN         NaN     NaN     NaN     NaN     NaN   
Genre         132          NaN         NaN     NaN     NaN     NaN     NaN   
Premiere        5          NaN         NaN     NaN     NaN     NaN     NaN   
Runtime       NaN    94.674464   26.648121     4.0    87.0    97.0   108.0   
IMDB Score    NaN     6.210916     0.96885     2.5     5.6     6.3     6.9   
Language      352          NaN         NaN     NaN     NaN     NaN     NaN   
is_original   513          NaN         NaN     NaN     NaN     NaN     NaN   

                 max  
Unnamed: 0     583.0  
show_id          NaN  
type             NaN  
title            NaN  
director         NaN  
cast             NaN  
country          NaN  
date_added       NaN  
release_year  2021.0  
rating           NaN  
duration         NaN  
listed_in        NaN  
description      NaN  
Title            NaN  
Genre            NaN  
Premiere         NaN  
Runtime        209.0  
IMDB Score       9.0  
Language         NaN  
is_original      NaN

count         mean         std     min     25%     50%     75%  \
Unnamed: 0    513.0   280.984405  166.445537     0.0   137.0   277.0   422.0   
release_year  513.0  2014.181287    9.995716  1959.0  2010.0  2019.0  2021.0   
Runtime       513.0    94.674464   26.648121     4.0    87.0    97.0   108.0   
IMDB Score    513.0     6.210916    0.968850     2.5     5.6     6.3     6.9   

                 max  
Unnamed: 0     583.0  
release_year  2021.0  
Runtime        209.0  
IMDB Score       9.0

In [43]:
print(df_netflix_edaproducciones.columns.tolist())  # ¿cuáles son nuestras columnas?

print("Análisis detallado de las columnas clave para la estrategia de contenidos")

display(df_netflix_edaproducciones["type"].unique())
display(df_netflix_edaproducciones["country"].unique())

['Unnamed: 0', 'show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added', 'release_year', 'rating', 'duration', 'listed_in', 'description', 'Title', 'Genre', 'Premiere', 'Runtime', 'IMDB Score', 'Language', 'is_original']
Análisis detallado de las columnas clave para la estrategia de contenidos


array(['Movie', 'TV Show'], dtype=object)

array(['United States', 'South Africa', <NA>, 'India',
       'United States, Ghana, Burkina Faso, United Kingdom, Germany, Ethiopia',
       'United Kingdom', 'Germany, Czech Republic', 'Mexico', 'Turkey',
       'Australia', 'United States, India, France', 'Finland',
       'China, Canada, United States',
       'South Africa, United States, Japan', 'Nigeria', 'Japan',
       'Spain, United States', 'France', 'Belgium',
       'United Kingdom, United States', 'United States, United Kingdom',
       'France, United States', 'South Korea', 'Spain',
       'United States, Singapore', 'United Kingdom, Australia, France',
       'United Kingdom, Australia, France, United States',
       'United States, Canada', 'Germany, United States',
       'South Africa, United States', 'United States, Mexico',
       'United States, Italy, France, Japan',
       'United States, Italy, Romania, United Kingdom',
       'Australia, United States', 'Argentina, Venezuela',
       'United States, United Ki

In [50]:
producciones_mejorvaloradas = df_netflix_edaproducciones[df_netflix_edaproducciones['IMDB Score'] > 8.5]
contenidos_peorvalorados = df_netflix_edaproducciones[df_netflix_edaproducciones['IMDB Score'] < 5]

print("¿Qué contenidos han alcanzado mayor éxito?")

print("Producciones mejor valoradas:")
display(producciones_mejorvaloradas)

print("\nLas 5 producciones peor valoradas:")
display((contenidos_peorvalorados.head(5)))

print("¿Cuánto influye la duración en la calificación obtenida?")
runtime_vs_score = df_netflix_edaproducciones[['Runtime', 'IMDB Score']].corr()
print(runtime_vs_score)

# La columna rating contiene las clasificaciones de audiencia asignadas a cada contenido, las cuales indican para qué público es adecuado

print("¿Cuántas producciones hay para cada clasificación de audiencia?")
ratings_distribution = df_netflix_edaproducciones['rating'].value_counts()
print("\nDistribución de Clasificaciones por Audiencia:")
print(ratings_distribution)

print("¿Y qué clasificaciones tienden a tener mejores calificaciones?")
imdb_by_rating = df_netflix_edaproducciones.groupby('rating')['IMDB Score'].mean().sort_values(ascending=False)
print("\nCalificación Promedio por Clasificación:")
print(imdb_by_rating)



¿Qué contenidos han alcanzado mayor éxito?
Producciones mejor valoradas:


Unnamed: 0 show_id   type                title      director  \
512         583    s584  Movie  No Strings Attached  Ivan Reitman   

                                                  cast  \
512  Natalie Portman, Ashton Kutcher, Kevin Kline, ...   

                           country    date_added  release_year rating  \
512  United States, United Kingdom  July 1, 2021          2011      R   

    duration                  listed_in  \
512  No data  Comedies, Romantic Movies   

                                           description  \
512  "Friends with benefits" Emma and Adam are fine...   

                                        Title        Genre         Premiere  \
512  David Attenborough: A Life on Our Planet  Documentary  October 4, 2020   

     Runtime  IMDB Score Language  is_original  
512       83         9.0  English         True


Las 5 producciones peor valoradas:


Unnamed: 0 show_id     type                  title         director  \
0           0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1           1      s2  TV Show          Blood & Water          Unknown   
2           2      s3  TV Show              Ganglands  Julien Leclercq   
3           3      s4  TV Show  Jailbirds New Orleans          Unknown   
4           4      s5  TV Show           Kota Factory          Unknown   

                                                cast        country  \
0                                            Unknown  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...           <NA>   
3                                            Unknown           <NA>   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA    No data   
3  September 24, 2021          2021  TV-MA    No data   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description            Title  \
0  As her father nears the end of his life, filmm...  Enter the Anime   
1  After crossing paths at a party, a Cape Town t...      Dark Forces   
2  To protect his family from a powerful drug lor...          The App   
3  Feuds, flirtations and toilet talk go down amo...   The Open House   
4  In a city of coaching centers known to train I...      Kaali Khuhi   

                   Genre           Premiere  Runtime  IMDB Score  \
0            Documentary     August 5, 2019       58         2.5   
1               Thriller    August 21, 2020       81         2.6   
2  Science fiction/Drama  December 26, 2019       79         2.6   
3        Horror thriller   January 19, 2018       94         3.2   
4                Mystery   October 30, 2020       90         3.4   

           Language  is_original  
0  English/Japanese         True  
1           Spanish         True  
2           Italian         True  
3           English         True  
4             Hindi         True

¿Cuánto influye la duración en la calificación obtenida?
             Runtime  IMDB Score
Runtime     1.000000   -0.055775
IMDB Score -0.055775    1.000000
¿Cuántas producciones hay para cada clasificación de audiencia?

Distribución de Clasificaciones por Audiencia:
rating
TV-MA    159
TV-14    108
PG-13     58
R         56
TV-PG     42
TV-Y7     35
PG        24
TV-Y      19
TV-G      11
G          1
Name: count, dtype: int64
¿Y qué clasificaciones tienden a tener mejores calificaciones?

Calificación Promedio por Clasificación:
rating
TV-G     6.418182
TV-14    6.326852
R        6.326786
TV-Y7    6.288571
TV-Y     6.200000
TV-MA    6.186164
PG-13    6.179310
PG       6.033333
TV-PG    5.900000
G        5.500000
Name: IMDB Score, dtype: float64


In [51]:
print("Análisis en clave estratégica de internacionalización")
print("Idiomas más representadados")
top_languages = df_netflix_edaproducciones['Language'].value_counts().head(10)
print("\nIdiomas Más Representados:")
print(top_languages)

print("Éxito de calificaciones según idioma")
imdb_by_language = df_netflix_edaproducciones.groupby('Language')['IMDB Score'].mean().sort_values(ascending=False).head(10)
print("\nIdiomas con Mejores Calificaciones Promedio:")
print(imdb_by_language)

Análisis en clave estratégica de internacionalización
Idiomas más representadados

Idiomas Más Representados:
Language
English            352
Hindi               28
Spanish             26
French              18
Italian             14
Portuguese           9
Indonesian           9
Japanese             5
English/Spanish      5
Korean               5
Name: count, dtype: int64
Éxito de calificaciones según idioma

Idiomas con Mejores Calificaciones Promedio:
Language
English/Ukranian/Russian    8.4
English/Akan                7.7
English/Hindi               7.3
English/Arabic              7.3
English/Mandarin            7.3
English/Russian             7.3
Spanish/English             7.3
English/Korean              7.3
Khmer/English/French        7.2
Tamil                       7.2
Name: IMDB Score, dtype: float64


# Clasificaciones más frecuentes 

TV-MA con 159 contenidos dirigido a audiencias adultas. Es la clasificación predominante en el catálogo, lo que indica que Netflix está enfocado en producciones para adultos. Esta cifra refleja el segmento estratégico para el mercado de contenidos de Netflix.
TV-14 con 108 contenidos: apto para mayores de 14 años, es la segunda clasificación más común, centrada en el target de adolescentes.
PG-13 con 58 contenidos: está orientado también a un público joven-adulto.
TV-G (11 contenidos) y G (1 contenido). Se trata de contenido apto para todas las edades. Estas categorías están subrepresentadas, lo que podría indicar una baja prioridad en el contenido infantil o familiar.

In [54]:
print("Años estrella por cantidadad de lanzamientos")
print("Número de contenido de estreno por año")
content_by_year = df_netflix_edaproducciones['release_year'].value_counts().sort_index()
print("\nContenido Agregado por Año:")
print(content_by_year)

print("Calificación promedio según su fecha de estreno en la plataforma")
avg_imdb_by_year = df_netflix_edaproducciones.groupby('release_year')['IMDB Score'].mean().sort_index()
print(avg_imdb_by_year)

Años estrella por cantidadad de lanzamientos
Número de contenido de estreno por año

Contenido Agregado por Año:
release_year
1959      1
1961      1
1975      1
1976      1
1978      1
1980      1
1982      1
1983      2
1984      1
1985      1
1986      3
1987      1
1989      1
1990      1
1991      3
1992      1
1993      6
1994      5
1995      5
1996      3
1997      4
1998      2
1999      6
2000      3
2001      5
2002      4
2003      9
2004      6
2005      5
2006      3
2007      7
2008     10
2009      8
2010     17
2011     13
2012     11
2013     15
2014     10
2015     12
2016     13
2017     14
2018     17
2019     33
2020     42
2021    204
Name: count, dtype: int64
Calificación promedio según su fecha de estreno en la plataforma
release_year
1959    7.600000
1961    5.800000
1975    4.700000
1976    7.400000
1978    4.700000
1980    5.800000
1982    5.600000
1983    5.950000
1984    5.800000
1985    6.400000
1986    6.000000
1987    4.800000
1989    5.600000
1990    5